DirectProgramming/C++SYCL_FPGA/Tutorials/GettingStarted/fpga_template/src/fpga_template.cpp

```diff
@@ -0,0 +1,114 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class VectorAddID;
+
+class VectorAdd {
```

Comment:
I agree that the constructor enforces argument ordering, but I think this design already has a lot of boiler-plate and I really want to do all I can to trim it down. I omit the default constructor because it adds clutter to what is supposed to be a concise design that shows the bare minimum required to create a oneAPI design on an FPGA.

Some other 'clutter' that was added during earlier revisions includes the SYCL host exception handler, and the frustrating device selector code.

Reply:
applied 'struct' conversion as per our discussion:

struct VectorAddID; struct VectorAdd {
  int *const a_in;
  int *const b_in;
  int *const c_out;
  int len;   void operator()() const {
    for (int idx = 0; idx < len; idx++) {
      int a_val = a_in[idx];
      int b_val = b_in[idx];
      int sum = a_val + b_val;
      c_out[idx] = sum;
    }
  }
};